controllers/ideas.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const path = require('path'),
+      models = require(path.resolve('./models')),
+      serialize = require(path.resolve('./serializers')).serialize;
+
+/**
+ * Create idea
+ */
+async function post(req, res, next) {
+  try {
+
+    // gather data
+    const { title, detail } = req.body;
+    const creator = req.auth.username;
+
+    // save the idea to database
+    const newIdea = await models.idea.create({ title, detail, creator });
```

Comment:
what if created idea is null? No 409?

Reply:
Can it be null? Can you think of a test when this would be the case?

Generally speaking, managing _duplicate_ ideas is still something to figure out.